### PR TITLE
[PM-4167] Add PRF attestation flow during passkey registration

### PR DESCRIPTION
**Pull request by coroiu**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Allow users to create and register all necessary keys to allow vault decryption using a passkey

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/Services/IUserService.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public interface IUserService
     Task<bool> DeleteWebAuthnKeyAsync(User user, int id);
     Task<bool> CompleteWebAuthRegistrationAsync(User user, int value, string name, AuthenticatorAttestationRawResponse attestationResponse);
     Task<CredentialCreateOptions> StartWebAuthnLoginRegistrationAsync(User user);
-    Task<bool> CompleteWebAuthLoginRegistrationAsync(User user, string name, CredentialCreateOptions options, AuthenticatorAttestationRawResponse attestationResponse);
+    Task<bool> CompleteWebAuthLoginRegistrationAsync(User user, string name, bool supportsPrf, string encryptedUserKey, string encryptedPublicKey, string encryptedPrivateKey, CredentialCreateOptions options, AuthenticatorAttestationRawResponse attestationResponse);
```
Comment (withinfocus) [inline src/Core/Services/IUserService.cs:31]: 🎨 Could these new parameters move to the end and be optional? For non-PRF default assumptions.

File: src/Core/Auth/Entities/WebAuthnCredential.cs [hunk 2]
```diff
@@ -29,4 +33,18 @@ public void SetNewId()
...
+        if (SupportsPrf && EncryptedUserKey != null && EncryptedPrivateKey != null && EncryptedPublicKey != null)
+        {
+            return WebAuthnPrfStatus.Enabled;
+        }
+        else if (SupportsPrf)
+        {
+            return WebAuthnPrfStatus.Supported;
+        }
+
+        return WebAuthnPrfStatus.Unsupported;
```
Comment (jlf0dev) [inline src/Core/Auth/Entities/WebAuthnCredential.cs:48]: nit (non-blocking): just a tiny suggestion for readability
```suggestion
        if (!SupportsPrf)
        {
            return WebAuthnPrfStatus.Unsupported;
        }
        if (EncryptedUserKey != null && EncryptedPrivateKey != null && EncryptedPublicKey != null)
        {
            return WebAuthnPrfStatus.Enabled;
        }
        return WebAuthnPrfStatus.Supported;

```

File: src/Core/Services/IUserService.cs (hunk 1 above)
Comment (jlf0dev) [reply src/Core/Services/IUserService.cs:31]: You could also consider making a data model for all these parameters and build it from the response model.

**Review by ike-kottlowski — APPROVED** [review]
Overall looks good. No notes.

**Review by jlf0dev — APPROVED** [review]

File: src/Core/Auth/Entities/WebAuthnCredential.cs (hunk 2 above)
Comment (coroiu) [reply src/Core/Auth/Entities/WebAuthnCredential.cs]: Fixed!

File: src/Core/Services/IUserService.cs (hunk 1 above)
Comment (coroiu) [reply src/Core/Services/IUserService.cs]: Fixed! I'll look into the model approach in my refactor task, where I'll convert these functions to commands!

**Pull request merged**